projects/kerttu-global/src/app/+validation/service/query-reset.guard.ts

```diff
@@ -0,0 +1,15 @@
+import { Injectable } from '@angular/core';
+import { ActivatedRouteSnapshot, CanActivate, RouterStateSnapshot } from '@angular/router';
+import { SpeciesListQueryService } from './species-list-query.service';
+
+@Injectable()
+export class QueryResetGuard implements CanActivate {
```

Comment:
The name is a bit generic, how about `SpeciesListQueryResetGuard`? That would help understanding the connection with the species list query  service.